virttest/utils_libvirt/libvirt_virtio.py

```diff
@@ -38,7 +37,7 @@ def add_iommu_dev(vm, iommu_dict):
     libvirt_vmxml.remove_vm_devices_by_type(vm, 'iommu')
     vmxml = vm_xml.VMXML.new_from_dumpxml(vm.name)
     features = vmxml.features
-    if not features.has_feature('ioapic') and arch.ARCH == "x86_64":
+    if not features.has_feature('ioapic') and iommu_dict.get('model') == "intel":
```

Comment:
How about AMD iommu? Do we need consider it as well?

Reply:
There is no AMD iommu device. On AMD host, we can also start a vm with intel iommu. In my understanding, it's not related with the host.

Reply:
@chunfuwen Could you please re-review this pr? Thx!

Reply:
I search AMD iommu and find one link https://www.amd.com/en/support/tech-docs/amd-io-virtualization-technology-iommu-specification. 
At glance, it looks like it is more of host, but not sure whether guest need align with host in this field.
But anyway, I approved it to move this PR forward.

Reply:
Thank you @chunfuwen for your sharing and reviewing!